platforms/hyperledger-indy/scripts/indy-ledger-script/indy-ledger.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+vault_addr=$1
+vault_token=$2
+admin_path=$3
+admin_name=$4
+identity_path=$5
+identity_name=$6
+identity_role=$7
+pool_genesis_path=$8
+
+[ -z "$vault_addr" ] && { echo "Script Failed: Vault Address Missing"; exit 1;} || echo "Vault Address Received";
+[ -z "$vault_token" ] && { echo "Script Failed: Vault Token Missing"; exit 1;} || echo "VAult Token Received";
+[ -z "$admin_path" ] && { echo "Script Failed: Admin DID Path Missing"; exit 1;} || echo "Admin DID Path Received";
+[ -z "$admin_name" ] && { echo "Script Failed: Admin Name Missing"; exit 1;} || echo "Admin Name Received";
+[ -z "$identity_path" ] && { echo "Script Failed: Identity DID Path Missing"; exit 1;} || echo "Identity DID Received";
+[ -z "$identity_name" ] && { echo "Script Failed: Identity DID Name Missing"; exit 1;} || echo "Identity Name Received";
+[ -z "$identity_role" ] && { echo "Script Failed: Identity DID Name Missing"; exit 1;} || echo "Identity Role Received";
+[ -z "$pool_genesis_path" ] && { echo "Script Failed: Genesis Pool File Path Missing"; exit 1;} || echo "Genesis Pool File Path Received";
+
+export VAULT_ADDR=$vault_addr
+export VAULT_TOKEN=$vault_token
+
+admin_did=$(vault kv get -field=did $admin_path/$admin_name)
+
+[ -z "$admin_did" ] && { echo "Script Failed"; exit 1;} || echo "Admin DID Acquired";
+admin_seed=$(vault kv get -field=seed $admin_path/$admin_name)
+[ -z "$admin_seed" ] && { echo "Script Failed"; exit 1;} || echo "Admin Seed Acquired";
+identity_did=$(vault kv get -field=did $identity_path/$identity_name)
+[ -z "$identity_did" ] && { echo "Script Failed"; exit 1;} || echo "DID Acquired";
+identity_seed=$(vault kv get -field=seed $identity_path/$identity_name)
+[ -z "$identity_seed" ] && { echo "Script Failed"; exit 1;} || echo "Seed Acquired";
+
+# Creating did files
+echo "{
+\"version\": 1,
+\"dids\": [{
+\"did\": \"$admin_did\", 
+\"seed\": \"$admin_seed\"
+}]
+}" > admindid.txt
+
+echo "{
+\"version\": 1,
+\"dids\": [{
+\"did\": \"$identity_did\", 
+\"seed\": \"$identity_seed\"
+}]
+}" > identitydid.txt
+
+echo "wallet create myIndyWallet key=12345
+wallet open myIndyWallet key=12345
+wallet list
+did import ./admindid.txt
+did import ./identitydid.txt
+did list
+exit" > indy_txn.txt
+
+indy-cli indy_txn.txt > txn_result.txt
+if grep -q 'Batch execution failed' 'txn_result.txt'
```

Comment:
I would rather have a check for a success message, I don't think you can cover all possible error messages

Reply:
The script gives a success message on a successful run.
On failure it does not check for any particular type of error,
It just fetches line line no at which the error is received and the reason for it.
Also I have included an exit code for failure.

Reply:
Please check just for the success message, it's safer.